homeassistant/components/modbus/__init__.py

```diff
@@ -6,39 +6,84 @@
 from pymodbus.transaction import ModbusRtuFramer
 import voluptuous as vol
 
+from homeassistant.components.cover import (
+    DEVICE_CLASSES_SCHEMA as COVER_DEVICE_CLASSES_SCHEMA,
+)
 from homeassistant.const import (
     ATTR_STATE,
+    CONF_COVERS,
     CONF_DELAY,
+    CONF_DEVICE_CLASS,
     CONF_HOST,
     CONF_METHOD,
     CONF_NAME,
     CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_SLAVE,
     CONF_TIMEOUT,
     CONF_TYPE,
     EVENT_HOMEASSISTANT_STOP,
 )
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
 
 from .const import (
     ATTR_ADDRESS,
     ATTR_HUB,
     ATTR_UNIT,
     ATTR_VALUE,
+    CALL_TYPE_COIL,
+    CALL_TYPE_REGISTER_HOLDING,
+    CALL_TYPE_REGISTER_INPUT,
     CONF_BAUDRATE,
     CONF_BYTESIZE,
+    CONF_INPUT_TYPE,
     CONF_PARITY,
+    CONF_REGISTER,
+    CONF_STATE_CLOSED,
+    CONF_STATE_CLOSING,
+    CONF_STATE_OPEN,
+    CONF_STATE_OPENING,
+    CONF_STATUS_REGISTER,
+    CONF_STATUS_REGISTER_TYPE,
     CONF_STOPBITS,
     DEFAULT_HUB,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_SLAVE,
     MODBUS_DOMAIN as DOMAIN,
     SERVICE_WRITE_COIL,
     SERVICE_WRITE_REGISTER,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-
 BASE_SCHEMA = vol.Schema({vol.Optional(CONF_NAME, default=DEFAULT_HUB): cv.string})
 
+COVERS_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CALL_TYPE_COIL, CONF_REGISTER),
+    vol.Schema(
+        {
+            vol.Required(CONF_NAME): cv.string,
+            vol.Optional(
+                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+            ): cv.positive_int,
```

Comment:
This should be cv.time_period as primary validator. Convert to seconds with a second validator.
```suggestion
            ): vol.All(cv.time_period, lambda value: value.total_seconds()),
```

Reply:
I applied your suggestion, and tested the change. Everything seems to be working fine.